go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -80,10 +85,10 @@ func (qre *QueryExecutor) Execute() (reply *sqltypes.Result, err error) {
 		tabletenv.RecordUserQuery(qre.ctx, qre.plan.TableName(), "Execute", int64(duration))
 
 		if reply == nil {
-			qre.plan.AddStats(1, duration, qre.logStats.MysqlResponseTime, 0, 1)
+			qre.addStats(planName, 1, duration, qre.logStats.MysqlResponseTime, 0, 1)
 			return
 		}
-		qre.plan.AddStats(1, duration, qre.logStats.MysqlResponseTime, int64(reply.RowsAffected), 0)
+		qre.addStats(planName, 1, duration, qre.logStats.MysqlResponseTime, int64(reply.RowsAffected), 0)
```

Comment:
This is somewhat stylistic, but IMO we should move the `qre.logStats` and `tabletenv.ResultStats` updating inside the new `qre.addStats` function to keep all the query metrics and log recording together.

Reply:
(Or don't have the function at all and just keep all the updating in-line... it's having a helper function that does some but not all of the stats work which I feel like could be cleaned up)